optimum/intel/ipex/inference.py

```diff
@@ -99,18 +185,18 @@ def __enter__(self):
                             with torch.cpu.amp.autocast(enabled=(self._dtype == torch.bfloat16)), torch.no_grad():
                                 if self._model.tokenizer is not None and self._jit:
                                     try:
-                                        jit_inputs = []
-                                        dummy_input = self._model.tokenizer("")
-                                        for key in dummy_input:
-                                            jit_inputs.append(torch.ones((1, len(dummy_input[key])), dtype=torch.long))
+                                        jit_inputs = prepare_jit_inputs(self._model.model, self._model.task)
                                         model = torch.jit.trace(model, jit_inputs, strict=False)
                                         model = torch.jit.freeze(model)
                                         model(*jit_inputs)
                                         model(*jit_inputs)
                                     except Exception as e:
                                         logger.warning(f"failed to use PyTorch jit mode due to: {e}.")
                                 # Patching model with the new one
-                                self._model.model = _ModelFallbackWrapper(model, self._original)
+                                if "text-generation" in self._model.task:

```

Comment:
Because we want to exclude the `"text2text-generation"` task, shouldn't it be :
```suggestion
                                if self._model.task == "text-generation":

```